tst/unit/test_meshblock_data_iterator.cpp

```diff
@@ -69,25 +70,30 @@ bool intervals_intersect(const std::pair<int, int> &i1, const std::pair<int, int
 TEST_CASE("Can pull variables from containers based on Metadata",
           "[MeshBlockDataIterator]") {
   GIVEN("A Container with a set of variables initialized to zero") {
-    MeshBlockData<Real> rc;
     std::vector<int> scalar_shape{16, 16, 16};
     std::vector<int> vector_shape{16, 16, 16, 3};
 
-    Metadata m_in({Metadata::Independent, Metadata::WithFluxes, Metadata::FillGhost},
-                  scalar_shape);
-    Metadata m_in_vector({Metadata::Independent, Metadata::WithFluxes,
-                          Metadata::FillGhost, Metadata::Vector},
+    Metadata m_in({Metadata::Independent, Metadata::WithFluxes}, scalar_shape);
+    Metadata m_in_vector({Metadata::Independent, Metadata::WithFluxes, Metadata::Vector},
                          vector_shape);
     Metadata m_out({Metadata::Derived}, scalar_shape);
     Metadata m_out_vector({Metadata::Derived}, vector_shape);
 
-    // Make some variables
-    rc.Add("v1", m_in);
-    rc.Add("v2", m_out);
-    rc.Add("v3", m_in_vector);
-    rc.Add("v4", m_out_vector);
-    rc.Add("v5", m_in);
-    rc.Add("v6", m_out);
+    // Make package with some variables
+    auto pkg = std::make_shared<StateDescriptor>("Test package");
+    pkg->AddField("v1", m_in);
+    pkg->AddField("v2", m_out);
+    pkg->AddField("v3", m_in_vector);
+    pkg->AddField("v4", m_out_vector);
+    pkg->AddField("v5", m_in);
+    pkg->AddField("v6", m_out);
+
+    // we need to connect the MeshBlockData to a dummy mesh block, otherwise variables
+    // won't be allocated
+    auto dummy_mb = std::make_shared<MeshBlock>(16, 3);
+
+    MeshBlockData<Real> rc;
+    rc.Initialize(pkg, dummy_mb);
```

Comment:
We could also simply use a non-default constructor with `std::make_shared` I would think.

Reply:
I think we need the default constructor for the `DataCollection` that contains the `MeshBlockData` for the different stages.

I would agree that a factory is a good pattern, but I don't think it's really need in this case, because the `MeshBlockData` instances are created internally and we only have to create some instances outside of the internal infrastructure inside the unit tests.